zipline/pipeline/loaders/events.py

```diff
@@ -0,0 +1,183 @@
+import abc
+import pandas as pd
+from six import iteritems
+from toolz import merge
+
+from .base import PipelineLoader
+from .frame import DataFrameLoader
+from .utils import next_date_frame, previous_date_frame, previous_value
+
+WRONG_COLS_ERROR = "Expected columns %s for sid %s but got columns %s."
+
+BAD_DATA_FORMAT_ERROR = ("Data for sid %s must be in DataFrame, "
+                         "Series, or DatetimeIndex.")
+
+SERIES_NO_DTINDEX_ERROR = ("Got Series for sid %d, but index was not "
+                           "DatetimeIndex.")
+
+DTINDEX_NOT_INFER_TS_ERROR = ("Got DatetimeIndex for sid %d.\n"
+                              "Pass `infer_timestamps=True` to use the first "
+                              "date in `all_dates` as implicit timestamp.")
+
+DF_NO_TS_NOT_INFER_TS_ERROR = ("Got DataFrame without a '%r' column for sid "
+                               "%d.\nPass `infer_timestamps=True` to use the "
+                               "first date in `all_dates` as implicit "
+                               "timestamp.")
```

Comment:
I'd like to test that these error messages are correctly raised, and also better style. Did you mean why are they specifically module-level rather than class-level?